genesis/process/shardGenesisBlockCreator.go

```diff
@@ -445,25 +445,38 @@ func deployInitialSmartContract(
 		return err
 	}
 
-	argDeploy := intermediate.ArgDeployProcessor{
-		Executor:       txExecutor,
-		PubkeyConv:     arg.PubkeyConv,
-		BlockchainHook: processors.blockchainHook,
-		QueryService:   processors.queryService,
-	}
-	deployProc, err := intermediate.NewDeployProcessor(argDeploy)
-	if err != nil {
-		return err
-	}
-
 	switch sc.GetType() {
 	case genesis.DelegationType:
 		deployMetrics.numDelegation++
-	default:
+		argDeploy := intermediate.ArgDeployProcessor{
+			Executor:       txExecutor,
+			PubkeyConv:     arg.PubkeyConv,
+			BlockchainHook: processors.blockchainHook,
+			QueryService:   processors.queryService,
+		}
+		deployProc, err := intermediate.NewDeployProcessor(argDeploy)
+		if err != nil {
+			return err
+		}
+
+		return deployProc.Deploy(sc)
+	case genesis.DNSType:
 		deployMetrics.numOtherTypes++
```

Comment:
should add numDnsType variable increment here.

Reply:
numDNS is only one - otherwise smart contract parser returns error.